src/test/FuzzerImpl.cpp

```diff
@@ -359,6 +360,16 @@ struct xdr_fuzzer_unpacker
         stellar::FuzzUtils::setShortKey(m.ed25519(), v);
     }
 
+    template <typename T>
+    typename std::enable_if<std::is_same<stellar::Asset, T>::value>::type
+    operator()(T& asset)
+    {
+        // 1 byte --> Asset
+        check(1);
+        std::uint8_t v = get_byte();
+        asset = stellar::FuzzUtils::makeAsset(v);
```

Comment:
`makeAsset` uses `strToAssetCode`, which will truncate the string to fit the `assetCode4`. Is it possible to terminate early and keep the search space to 10 assets? Or would it just be better to use an `ASSET_TYPE_CREDIT_ALPHANUM12` here?

Reply:
Yes, you're right. I'm not sure if that's a good idea though, just because if we were to increase the number of assets this code would cease to work.

Reply:
Maybe we should start with something simple and stupid like using the modulus of the byte with something like 9, with one result meaning native asset and the others meaning `"Ast<N>"`?  It'd be one digit, and since we create 4 assets during setup, the non-native assets that we'd end up referring to would definitely exist about half the time and not-necessarily-exist about half the time.  We could in later PRs add things like:

- Sometimes using invalid assets, maybe a fixed couple at first, maybe later deterministically generated from certain bytes
- Sometimes using 12-byte asset codes

Another possible future enhancement relates to the `makeValid()` functions that we already have for `TrustLineEntry` and `OfferEntry` not using the randomly-generated `AssetCode` at all -- they just use fixed assets such as `"USD"`, `"CAD"`, and `"EUR"`.  Adapting them to use the generated `AssetCode` that's passed into them might produce code that could be shared by the fuzzer.

Reply:
I filed https://github.com/stellar/stellar-core/issues/2924 for potential future enhancements.